lib/protocol/operator/004/operator_004.ts

```diff
@@ -196,13 +196,26 @@ export class SNProtocolOperator004 extends SNProtocolOperator003 {
   private async authenticatedDataToString(
     attachedData: ItemAuthenticatedData
   ) {
-    return this.crypto.base64Encode(JSON.stringify(sortedCopy(attachedData)));
+    return this.crypto.base64Encode(
+      JSON.stringify(
+        sortedCopy(
+          omitUndefinedCopy(
```

Comment:
Technically JSON.stringify will already omit any keys for which the value is undefined. But I didn't want to count on this being an implementation detail of web browsers. Perhaps React Native does it differently. So this is why we omit undefined keys explicitly.